src/dir.targets

```diff
@@ -174,6 +188,33 @@
     <Error Condition="'$(TestRunExitCode)' != '0'" Text="One or more tests failed while running tests from '$(MSBuildProjectName)' please check log for details!" />
   </Target>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(TestDisabled)'!='True'">
+    <PropertyGroup>
+      <!-- By default project has only inner-loop tests. -->
+      <TestCategories Condition="'$(TestCategories)'==''">InnerLoop</TestCategories>
+    </PropertyGroup>
+    <ItemGroup>
```

Comment:
This target can simply become a filtering of the TestCategories via the RunTestWithTraits item if you switch to that model.